### DISCUSSION TITLE
Issue with API Versioning: All APIs Merged into Single V1Resource.java Class

### DISCUSSION THREAD

**Discussion by thongdanghoang** [quarkiverse/quarkus-openapi-generator#876] [Q&A]
When using the quarkus-openapi-generator-server to generate server code from the OpenAPI YAML specification, all API paths with the same versioning prefix (e.g., /v1/...) are merged into a single resource class (V1Resource.java). This behavior is problematic because it does not reflect the intended grouping of endpoints by their respective tags (e.g., InventoriesResource, PermissionsResource).

For example, the following OpenAPI YAML:

```yaml
openapi: 3.0.3
info:
  title: Example API
  version: 1.0.0
paths:
  /v1/inventory:
    get:
      tags:
        - InventoriesResource
      summary: Retrieve inventories
      ...
  /v1/permissions/inventories:
    get:
      tags:
        - PermissionsResource
      summary: Retrieve permissions
      ...
```

My Quarkus application.yml:
```yaml
    openapi:
        generator:
            base-package: ch.admin.wbf.digiflux.stocks.rest
            spec: publicapi.yaml
```
And quarkus-openapi-generator-server versio:
```yaml
<quarkus-openapi-generator-server.version>2.6.0-lts</quarkus-openapi-generator-server.version>
```

Generates a single V1Resource.java class instead of separate classes (InventoriesResource.java and PermissionsResource.java).

**Expected Behavior:**

The generator should create separate resource classes based on the tags property in the OpenAPI YAML. For the example above, it should generate:

InventoriesResource.java containing /v1/inventory-related endpoints.
PermissionsResource.java containing /v1/permissions/inventories-related endpoints.
**Actual Behavior:**

All endpoints with the /v1/... prefix are merged into a single class: V1Resource.java.

**Steps to Reproduce:**

Create an OpenAPI YAML specification with versioned paths (e.g., /v1/...).
Use the quarkus-openapi-generator-server to generate the server code.
Observe that all endpoints are grouped into a single resource class based on the version prefix.

**Comment by ricardozanini** [comment]
cc @carlesarnal can you please take a look?

**Reply by clemthi** [reply]
Hello @ricardozanini @carlesarnal 
were you able to check into this issue?
I think that if the resources would be generated using the entire path it would be acceptable.
If we take @thongdanghoang example, we would have something like `V1InventoryRelatedResource` and `V1PermissionsInventoriesRelatedResource`.

**Reply by ricardozanini** [reply]
@mcruzdev can you take a look?

**Reply by mcruzdev** [reply]
I agree with you @clemthi. I created an issue on the apicurio-codegen project https://github.com/Apicurio/apicurio-codegen/issues/336. I already have a soluition.

**Reply by aservedio-swi** [reply]
@mcruzdev - not suggesting every generator should generate the same thing, but I have observed that the `org.openapitools:openapi-generator-maven-plugin` maven plugin with `configOptions` `useTags` set to `true` would generate classes `InventoriesResource.java` and `PermissionsResource.java` for the example above. I suppose they do that instead of `V1InventoriesResource.java` and `V1PermissionsResource.java` because people could technically have a mix of path prefixes within related to a tag.

Meaning, I think the tag is a way for people to gather related API's into a single implementation as they see fit with a name for that class that makes sense to their use case.

That is just one point of view of course...
